### PR TITLE
Backport 1-1: Update docs for new consensus settings

### PR DESCRIPTION
**Pull request by danintel**
Updates documentation to include information about the new consensus
engine settings, `sawtooth.consensus.algorithm.name` and
`sawtooth.consensus.algorithm.version`, and remove erroneous
information.

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

**Comment by danintel** [comment]
**Background notes**
The nightly documentation was updated with the new consensus engine settings on 2018-10-10 with this push:
https://github.com/hyperledger/sawtooth-core/commit/6a1d3bf2653a3660ec4b016b75e94a789f89dbed#diff-cd6bf86169fbdf3de3325c36e1640deb

Some, *but not all*, of these changes were backported to the 1-1 latest docs with this push on 2019-10-23 with this push:
https://github.com/hyperledger/sawtooth-core/commit/f4649ccb3ba6bab4055be06334811912d2a8fd73#diff-65a6136e31afa8b3258bb88c56f6dd40
That is, these two files were updated:
`docs/source/sysadmin_guide/creating_genesis_block.rst`
`docs/source/sysadmin_guide/setting_allowed_txns.rst`

However, three files were *not* backported:
`docs/source/app_developers_guide/creating_sawtooth_network.rst`
`docs/source/architecture/journal.rst`
`docs/source/sysadmin_guide/configure_sgx.rst`

This PR completes the backport by updating these three missing files.

**Comment by danintel** [comment]
This PR was motivated by a plea on Rocket.Chat on 2019-03-08 that the documentation "simply doesn't work":

> This PR was motivated by a plea on Rocket.Chat that the documentation "simply doesn't work":
>  Daniele 35 6:03 AM
> Hello, I'm a newbie in sawtooth. I started with the basic example, with single validator node, experimenting with the "tuna" demo (for example) but now I would like to work with a "real" blockchain network or at least, let's say, a multi-node network to run with docker-compose on my laptop. I followed the instruction on the official site:
> https://sawtooth.hyperledger.org/docs/core/releases/latest/app_developers_guide/creating_sawtooth_network.html#about-the-sawtooth-network-environment
> but there's a problem with the provided docker-compose yaml.... it simply doesn't work: you can start the network and see the nodes exchanging peers information and register one to each other, even exchanging a first block, but few seconds afterwards all nodes were disconnecting from the "validator-0" (with log messages like "Removed send_message function for connection..." for each node). So the network is not working (tried intkey processor for example).
> Any Idea about the reason? The docker-compose yaml I used is the official one:
> https://sawtooth.hyperledger.org/docs/core/releases/latest/app_developers_guide/sawtooth-default-poet.yaml
> Does anyone have a working example of a real multi-node network with PoET consensus? I couldn't find anything...
> >

File: docs/source/sysadmin_guide/configure_sgx.rst [hunk 1]
```diff
@@ -259,9 +260,12 @@ lines of output showing that the SGX enclave has been initialized:
     There’s quite a bit going on in the previous ``sawset proposal`` command, so
     let’s take a closer look at what it accomplishes:
 
-    ``sawtooth.consensus.algorithm=poet``
+    ``sawtooth.consensus.algorithm=PoET``
```
Comment (agunde406) [inline docs/source/sysadmin_guide/configure_sgx.rst]: This should be sawtooth.consensus.algorithm.name

File: docs/source/architecture/journal.rst [hunk 2]
```diff
@@ -310,8 +311,7 @@ If any transactions in the genesis block set or change settings, Sawtooth
 requires the `Sawtooth Settings transaction processor <../cli/settings-tp>`_
 or an equivalent implementation.
 For example, if the genesis block configures PoET consensus, this transaction
-processor is handles the transactions with PoET settings. (If no
-consensus is specified, Sawtooth uses dev mode consensus.)
+processor is handles the transactions with PoET settings.
```
Comment (agunde406) [inline docs/source/architecture/journal.rst]: Can you also remove the "is" from the previous sentence?

File: docs/source/sysadmin_guide/configure_sgx.rst (hunk 1 above)
Comment (danintel) [reply docs/source/sysadmin_guide/configure_sgx.rst]: Fixed.

File: docs/source/architecture/journal.rst (hunk 2 above)
Comment (danintel) [reply docs/source/architecture/journal.rst]: I saw that. The mistake is still in the master branch, so I left it there.

So I fixed it. The error remains in the master branch.

**Comment by danintel** [comment]
Please review. It's a backport of missing documentation corrections.

**Pull request merged**